applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -170,8 +170,19 @@ Vector& ElasticIsotropic3D::CalculateValue(
     Vector& rValue
     )
 {
-    if (rThisVariable == STRAIN) {
+    if (rThisVariable == STRAIN || 
+        rThisVariable == GREEN_LAGRANGE_STRAIN_VECTOR ||
+        rThisVariable == ALMANSI_STRAIN_VECTOR) {
         this->CalculateCauchyGreenStrain( rParameterValues, rValue);
+    } else if (rThisVariable == STRESSES || 
+        rThisVariable == CAUCHY_STRESS_VECTOR ||
+        rThisVariable == KIRCHHOFF_STRESS_VECTOR ||
+        rThisVariable == PK2_STRESS_VECTOR) {
+        // Get Values to compute the constitutive law:
+        Flags& r_flags = rParameterValues.GetOptions();
+        r_flags.Set( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false );
+        r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, true );
+        this->CalculateMaterialResponseCauchy(rParameterValues);
```

Comment:
rValue is not provided in this case. Is it ok?

Reply:
This the infinitesimal strain law, which can be used with both large and small displacement element. I am assuming that this configuration has been set from the element, but maybe there is a better option

Reply:
I am thinking that maybe the proper thing to do is to save the flags before setting and setting again after computing, otherwise could affect computation if called inside other law

Reply:
I am not sure about the proper place where the flags should be set. If they may be set anywhere, so it will be a good practise to save them before to later restore them.